human_cancer_analysis/nbconverted/2_identify_generic_genes_pathways.py

```diff
@@ -0,0 +1,778 @@
+# * We want to compare the ranking of pathways identified using the above method with the ranking found from Powers et. al., which identified a set of pathways as generic based on their ranking;
+# * This comparison will validate our method being used as a way to automatically identify generic genes and pathways.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('load_ext', 'rpy2.ipython')
+get_ipython().run_line_magic('autoreload', '2')
+
+import os
+import sys
+import pandas as pd
+import numpy as np
+import pickle
+
+from rpy2.robjects import pandas2ri
+pandas2ri.activate()
+
+from ponyo import utils
+from generic_expression_patterns_modules import calc, process
+
+np.random.seed(123)
+
+
+# In[2]:
+
+
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_human_cancer.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+
+# In[3]:
+
+
+# Load params
+local_dir = params["local_dir"]
+dataset_name = params['dataset_name']
+NN_architecture = params['NN_architecture']
+num_runs = params['num_simulated']
+project_id = params['project_id']
+metadata_col_id = params['metadata_colname']
+processed_template_filename = params['processed_template_filename']
+normalized_compendium_filename = params['normalized_compendium_filename']
+scaler_filename = params['scaler_filename']
+col_to_rank_genes = params['rank_genes_by']
+col_to_rank_pathways = params['rank_pathways_by']
+compare_genes = params['compare_genes']
+statistic = params['gsea_statistic']
+
+# Load metadata file with grouping assignments for samples
+sample_id_metadata_filename = os.path.join(
+    base_dir,
+    dataset_name,
+    "data",
+    "metadata",
+    f"{project_id}_process_samples.tsv"
+)
+
+# Load pickled file
+with open(scaler_filename, "rb") as scaler_fh:
+    scaler = pickle.load(scaler_fh)
+
+
+# In[4]:
+
+
+# Output files
+gene_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_gene_summary_{project_id}.tsv"
+)
+
+pathway_summary_filename = os.path.join(
+    base_dir, 
+    dataset_name, 
+    f"generic_pathway_summary_{project_id}.tsv"
+)
+
+
+# ## Need to customize code from ponyo
+# 
+# The current simulation-related function in ponyo, `get_sample_ids` assumes that the user is using one of two different metadata files (one associated with the pseudomonas compendium and another associated with recount2). The compendium dataset we are using here has a slightly different format for their metadata file. 
+# 
+# Here we are temporarily writing our own function customized for this Powers et. al. dataset. But we will be updating ponyo to allow for different metadata files in the future. Issue in ponyo is [here](https://github.com/greenelab/ponyo/issues/18) 
+
+# In[5]:
+
+
+def get_sample_ids(experiment_id, dataset_name, sample_id_colname):
```

Comment:
Maybe you could just have this function take the metadata file as an argument? Then it seems like this function would generalize to the cases you have in `ponyo` as well, but then the user has to provide the file path of the metadata, so it's a bit of a tradeoff. Up to you.

Reply:
Yes i think its better to use the metadata file as input and what I plan to do when I update ponyo.